### PR TITLE
build documentation and build script improvement

### PR DESCRIPTION
**Pull request by aerickson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,20 @@ User can switch the open type in KUAL KPVBooklet extentions.
 
 KPVBooklet is licensed under the MIT license. See the file
 LICENSE for more details.
+
+## building
+
+- gather and install dependencies
+  - getting kindle device jars
```
Comment (Frenzie) [inline README.md]: What is that for exactly?

Comment (aerickson) [reply README.md]: They're dependencies for the app. The ant script won't run without them present.

Comment (Frenzie) [reply README.md]: And these Kindle device jars are proprietary I assume? That would probably preclude CI.

Comment (NiLuJe) [reply README.md]: Yes they are, as well as potentially FW & model-dependent in some circumstances.

**Comment by NiLuJe** [comment]
I haven't looked at the antfile, because I've never played with KPV, but you also need a Java version that can output 1.4 byte code.

Historically, people where using a tremendously old JDK that matched the Kindle's roughly, but I'm personally using the Oracle JDK w/ the right compile flags, flags which have been dropped in Java 9.

File: README.md (hunk 1 above)
Comment (NiLuJe) [reply README.md]: In some circumstances (not familiar with KPV, don't know if it's affected), some of the private APIs from these JARs we were relying on have been obfuscated in newer FW, which makes supporting them much harder.

**Comment by NiLuJe** [comment]
Checked the antfile, it's properly requesting 1.8 bytecode, so that'll work with Oracle's JDK, too ;).

**Comment by NiLuJe** [comment]
For science, I wouldn't mind you checking whether an OpenJDK 8 build with source/target set to 1.8 works on your device ;).

**Comment by aerickson** [comment]
The 1.4 change is required when building with OpenJDK 1.8.

Output from trying this PR with the build.xml set to 1.8:

```
140101:120706 appmgrd[1064]: I appmgr_registry:app_found_for_uri:lipc_app_id=com.github.koreader.kpvbooklet,executable=NONE:
140101:120706 appmgrd[1064]: I appmgr_history:NEW_APP:lipcId=com.github.koreader.kpvbooklet:
140101:120706 cvm[7638]: I BookletManager:StartingBooklet:booklet=com.github.koreader.kpvbooklet:
140101:120706 appmgrd[1064]: I appmgr_history:app_state_change:lipc_id=com.github.koreader.kpvbooklet,state=EXECUTED:
140101:120706 cvm[7638]: E BookletManager:BookletClassInitError:class=/opt/amazon/ebook/booklet/KPVBooklet.jar:
140101:120706 cvm[7638]: java.lang.UnsupportedClassVersionError: com/github/chrox/kpvbooklet/KPVBooklet (Unsupported major.minor version 52.0)
140101:120706 cvm[7638]: 	at java.lang.ClassLoader.defineClass0(Native Method)
140101:120706 cvm[7638]: 	at java.lang.ClassLoader.defineClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.security.SecureClassLoader.defineClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.net.URLClassLoader.defineClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.net.URLClassLoader.access$100(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.net.URLClassLoader$1.run(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.security.AccessController.doPrivileged(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.net.URLClassLoader.findClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at sun.misc.Launcher$AppClassLoader.findClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.lang.ClassLoader.loadClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at sun.misc.Launcher$AppClassLoader.loadClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.lang.ClassLoader.loadClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at java.lang.ClassLoader.loadClass(Compiled Method)(Unknown Source)
140101:120706 cvm[7638]: 	at com.amazon.kindle.restricted.booklet.impl.BookletManager.Tnb(aqc:827)
140101:120706 cvm[7638]: 	at com.amazon.kindle.restricted.booklet.impl.C.run(aqc:1537)
140101:120706 cvm[7638]: 	at com.amazon.ebook.util.b.a.run(ftf:252)
140101:120706 cvm[7638]: 	at java.lang.Thread.run(Unknown Source)
140101:120706 cvm[7638]: 	at java.lang.Thread.startup(Unknown Source)
140101:120706 pillow[7268]: I def:show-system-alert:id=appmgrAppFailedFatal, altId=appmgrAppFailedFatal, alertLogInfo=undefined:
140101:120706 appmgrd[1064]: I appmgr_lipc:set_appmgr_run_result:propName=runresult,buf=1:com.github.koreader.kpvbooklet:entry
140101:120706 appmgrd[1064]: I appmgr_history:app_state_change:lipc_id=com.github.koreader.kpvbooklet,state=REGISTERED:
140101:120706 appmgrd[1064]: E lipc:dbuserr:name=org.freedesktop.DBus.Error.ServiceUnknown:Receive Dbus error message (DBUS_MESSAGE_TYPE_ERROR): The name com.github.koreader.kpvbooklet was not provided by any .service files
140101:120706 appmgrd[1064]: E appmgr_lipc:lipc_set_prop_fail:lipc_status=3,app=com.github.koreader.kpvbooklet,property=load:send app command
140101:120706 todo[1057]: I todo:ProcessingToDo:status=starting,reason=NetworkStartup:Started processing ToDo
140101:120706 appmgrd[1064]: I appmgr_lipc:default_alert_displayed:app_lipc_id=com.github.koreader.kpvbooklet:show_simple_alert
140101:120706 cvm[7638]: I BookletManager:TerminatingBooklet:booklet=com.github.koreader.kpvbooklet:
140101:120706 cvm[7638]: I lipc:evts:name=credentialsInstalled, origin=com.lab126.DeviceAuthenticationService:Event sent
140101:120706 todo[1057]: W url_builder:GetURLWarning:reason=missingGlobalParameter,param=currentCountry:
140101:120706 todo[1057]: E todo_get_document:GetToDoFailed:reason=ensureConnectionFailed,error=5:
140101:120706 todo[1057]: E todo:ProcessingToDoFailed:reason=getDocumentFailedNullDocument,errno=0:Null ToDo document returned
[root@kindle root]# 

```

**Comment by NiLuJe** [comment]
:+1:

Thanks for going to the trouble of confirming that ;).

**Comment by aerickson** [comment]
@NiLuJe I can't land this change, will you or would you prefer @chrox do it?

Thanks. :)

**Comment by chrox** [comment]
@aerickson Thanks for the patch. :)

**Pull request merged**